ui/src/composables/use-cached-filters.js

```diff
@@ -0,0 +1,74 @@
+import useContext from './use-context'
+import { SessionStorage } from 'quasar'
+import { reactive } from 'vue'
+import { useRouter, useRoute } from 'vue-router'
+
+const cachedFilters = reactive({})
+
+export default function (entity) {
+  function updateSessionStorage () {
+    SessionStorage.set('cachedFilters', cachedFilters)
+  }
+
+  async function init () {
+    const router = useRouter()
+    const route = useRoute()
+
+    cachedFilters[entity] = {}
+
+    const { search, filters } = useContext().context.value
+
+    const hasQueryParams = search || Object.keys(filters).length
+
+    if (hasQueryParams) {
+      search && addOne({ label: 'search', value: search })
+      Object.keys(filters).forEach(filter => addOne({ label: filter, value: filters[filter] }))
+    } else {
+      const storedFilters = SessionStorage.getItem('cachedFilters') || {}
+      Object.keys(storedFilters).forEach(filter => (cachedFilters[filter] = storedFilters[filter]))
+    }
+
+    await router.push({ query: { ...route.query, ...cachedFilters[entity] } })
```

Comment:
Só para eu entender, se o usuário já vier com alguma query na rota, você não vai usar o cache, certo?

Se for isso, eu acho que esse `router.push` só deveria estar dentro do `else` não? 🤔

Reply:
A query terá preferência em relação ao cache